test/Microsoft.AspNetCore.Rewrite.Tests/UrlRewrite/FormatExceptionHandlingTests.cs

```diff
@@ -0,0 +1,119 @@
+
+namespace Microsoft.AspNetCore.Rewrite.Tests.UrlRewrite
+{
+    public class FormatExceptionHandlingTests
+    {
+        [Theory]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <action type=""Rewrite"" url =""{"" />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <action type=""AbortRequest"" url ="""" />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <action type=""CustomResponse"" url ="""" />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <action type=""CustomResponse"" />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <conditions>  
+                <add input=""{HTTPS"" pattern=""^OFF$"" />  
+            </conditions>  
+            <action type=""Rewrite"" url =""foo"" />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <conditions>  
+                <add pattern=""^OFF$"" />  
+            </conditions>  
+            <action type=""Rewrite"" url =""foo"" />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <conditions>  
+                <add input=""{HTTPS}"" />  
+            </conditions>  
+            <action type=""Rewrite"" url =""foo"" />
+        </rule>
+    </rules>
+</rewrite>")]
+        [InlineData(
+@"<rewrite>
+    <rules>
+        <rule name=""Rewrite to article.aspx"">
+            <match url = ""(.*)"" />
+            <conditions>  
+                <add input=""{HTTPS}"" patternSyntax=""ExactMatch""/>  
+            </conditions>  
+            <action type=""Rewrite"" url =""foo"" />
+        </rule>
+    </rules>
+</rewrite>")]
+        public void ThrowFormatExceptionWithCorrectMessage(string input)
+        {
+            // Arrange, Act, Assert
+            Assert.Throws<FormatException>(() => UrlRewriteFileParser.Parse(new StringReader(input)));
```

Comment:
In this test it makes sense to check because you're testing several different kinds of failures.